src/lpc.jl

```diff
@@ -0,0 +1,75 @@
+"""
+`lpc_burg(x::Vector, p::Int)`
+
+LPC (Linear-Predictive-Code) estimation, using the Burg method.
+
+Given input signal `x` and prediction order `p`, returns IIR coefficients `a`
+and average reconstruction error `prediction_err`. Note that this method does
+NOT return the leading `1` present in the true autocorrelative estimate; it
+omits it as it is implicit in every LPC estimate, and must be manually
+reintroduced if the returned vector should be treated as a polynomial. This
+function implements the mathematics published in [1].
+
+[1] - Enhanced Partial Tracking Using Linear Prediction
+(DAFX 2003 article, Lagrange et al)
+http://www.sylvain-marchand.info/Publications/dafx03.pdf
+"""
+function lpc_burg{T <: Number}(x::Vector{T}, p::Int)
+    ef = x                      # forward error
+    eb = x                      # backwards error
+    a = [1; zeros(T, p)...]     # prediction coefficients
+
+    # Initialize prediction error wtih the variance of the signal
+    prediction_err = (x'*x ./ length(x))[1]
+
+    for m in 1:p
+        efp = ef[1:end-1]
+        ebp = eb[2:end]
```

Comment:
Not sure if performance is a concern here, but if I am reading the algorithm right, this could be done in-place by copying `ef` and `eb` into vectors at the start of the function, and then using `shift!` and `pop!` along with in-place scaling below.